src/core/classes/items-management/routed-items-manager.ts

```diff
@@ -152,6 +152,11 @@ export abstract class CoreRoutedItemsManager<
      */
     protected onSourceItemsUpdated(items: Item[]): void {
         super.onSourceItemsUpdated(items);
+        const selectedItem = this.selectedItem;
+
+        if (selectedItem !== null && items.some(item => this.getItemId(item) === this.getItemId(selectedItem))) {
```

Comment:
```suggestion
        if (selectedItem !== null && items.some(item => item === selectedItem)) {
```

We also need to compare by reference rather than id here. I've reviewed the code to make sure we're not forgetting it anywhere else, and I think it should be fine after fixing this line.